### PR TITLE
futurefix: fix a deprecation warning from mpl 3.3

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

the warning in question is seen in Jenkins logs, e.g.
```
/home/fido/workspace/yt_py38_git/yt/visualization/base_plot_types.py:274:
MatplotlibDeprecationWarning: Passing parameters norm and vmin/vmax simultaneously is deprecated since 3.3 
and will become an error two minor releases later. Please pass vmin/vmax directly to the norm when creating it.
  self.image = self.axes.imshow(
```

### CODE REVIEW

File: yt/visualization/base_plot_types.py [hunk 1]
```diff
@@ -217,34 +217,43 @@ def __init__(self, fsize, axrect, caxrect, zlim, figure, axes, cax):
 
     def _init_image(self, data, cbnorm, cblinthresh, cmap, extent, aspect):
         """Store output of imshow in image variable"""
+        cbnorm_kwargs = dict(
+            vmin=float(self.zmin) if self.zmax is not None else None,
```
Comment (neutrinoceros) [inline yt/visualization/base_plot_types.py:221]:
```suggestion
            vmin=float(self.zmin) if self.zmin is not None else None,
```

**Comment by neutrinoceros** [comment]
This now also includes a fix for a bug that was due to overlooked copy-pasted code.

File: yt/visualization/base_plot_types.py [hunk 2]
```diff
@@ -217,34 +217,43 @@ def __init__(self, fsize, axrect, caxrect, zlim, figure, axes, cax):
...
-            symlog_kwargs = dict(
-                linthresh=cblinthresh,
-                vmin=float(np.nanmin(data)),
-                vmax=float(np.nanmax(data)),
+            cbnorm_kwargs.update(
+                dict(
+                    linthresh=cblinthresh,
+                    vmin=float(np.nanmin(data)),
+                    vmax=float(np.nanmax(data)),
+                )
```
Comment (cphyc) [inline yt/visualization/base_plot_types.py:237]: In my dream where we are already all using Python 3.9
```suggestion
            cbnorm_kwargs |= dict(
                linthresh=cblinthresh,
                vmin=float(np.nanmin(data)),
                vmax=float(np.nanmax(data)),
            )
```

File: yt/visualization/base_plot_types.py [hunk 3]
```diff
@@ -217,34 +217,43 @@ def __init__(self, fsize, axrect, caxrect, zlim, figure, axes, cax):
...
             )
             MPL_VERSION = LooseVersion(matplotlib.__version__)
             if MPL_VERSION >= "3.2.0":
                 # note that this creates an inconsistency between mpl versions
                 # since the default value previous to mpl 3.4.0 is np.e
                 # but it is only exposed since 3.2.0
-                symlog_kwargs.update(dict(base=10))
+                cbnorm_kwargs.update(dict(base=10))
```
Comment (cphyc) [inline yt/visualization/base_plot_types.py:244]: This could be reworded as
```suggestion
                cbnorm_kwargs["base"] = 10
```

**Review by cphyc — APPROVED** [review]
I left one comment, you can decide if you want it in by accepting it, or simply ignore it by marking the discussion as resolved. Either way, I'm ok with merging the PR!

File: yt/visualization/base_plot_types.py (hunk 2 above)
Comment (neutrinoceros) [reply yt/visualization/base_plot_types.py:237]: Yeah I actually wrote this _while listening to a podcast on Python 3.9.0_... I had to resist the urge of doing exactly this :D

File: yt/visualization/base_plot_types.py (hunk 3 above)
Comment (neutrinoceros) [reply yt/visualization/base_plot_types.py:244]: I agree this is more clear than the original code, thanks.

**Review by Xarthisius — APPROVED** [review]

**Review by munkm — APPROVED** [review]

**Pull request merged**